### ISSUE TITLE
Forward-port broken if intermediate commit requires `renamelimit=0`

### ISSUE THREAD

**Issue by xmo-odoo** [odoo/runbot#444]
https://github.com/odoo/odoo/pull/63290 only has one commit despite its source having 5, this is not because the first 4 commits were suppressed due to being backports.

Issue is:

https://github.com/odoo/runbot/blob/7ab7eed27e985a78ce302aee584a362933eb829d/forwardport/models/project.py#L894-L900

if the cherrypick fails, we reset to `original_head` which is the head of the branch we're rebasing on, so if the commit being cherrypicked is not the first and the cherrypick ultimately succeeds every preceding commit is disappeared.

**Comment by xmo-odoo** [comment]
Fix is probably just adding a `prev = original_head = ...` then `prev = new = ...` at the end of the loop, and resetting to `prev` instead of `original_head`at this location.

Also might be useful to add some sort of separator at the end of the debug call? as-is the log looks a bit... odd:
```
2020-12-14 07:39:47,284 8208 DEBUG mergebot odoo.addons.forwardport: Cherry-picked e5ccb69f7a08fe5a9a66fc2888448dc2b5428f8f: 
128

fatal: inflateInit: out of memory (no message)
 
2020-12-14 07:39:49,165 8208 DEBUG mergebot odoo.addons.forwardport: Cherry-picked e5ccb69f7a08fe5a9a66fc2888448dc2b5428f8f (
renamelimit=0): 0
[master-14.0-fix-fonts-moc-XAA2-fw 9e8db65266d] [FIX] web: Allow accented uppercase characters in sign
 Date: Thu Nov 19 15:53:48 2020 +0100
 1 file changed, 1 insertion(+), 2 deletions(-)

Performing inexact rename detection:  65% (370524/563004)   ^MPerforming inexact rename detection:  66% (372930/563004)   ^MP
erforming inexact rename detection:  67% (377742/563004)   ^MPerforming inexact rename detection:  68% (384960/563004)   ^MPe
rforming inexact rename detection:  69% (389772/563004)   ^MPerforming inexact rename detection:  70% (394584/563004)   ^MPer
forming inexact rename detection:  71% (404208/563004)   ^MPerforming inexact rename detection:  72% (409020/563004)   ^MPerf
orming inexact rename detection:  73% (411426/563004)   ^MPerforming inexact rename detection:  74% (418644/563004)   ^MPerfo
rming inexact rename detection:  75% (425862/563004)   ^MPerforming inexact rename detection:  76% (428268/563004)   ^MPerfor
ming inexact rename detection:  78% (440298/563004)   ^MPerforming inexact rename detection:  79% (447516/563004)   ^MPerform
ing inexact rename detection:  80% (452328/563004)   ^MPerforming inexact rename detection:  81% (459546/563004)   ^MPerformi
ng inexact rename detection:  82% (461952/563004)   ^MPerforming inexact rename detection:  83% (469170/563004)   ^MPerformin
g inexact rename detection:  84% (473982/563004)   ^MPerforming inexact rename detection:  85% (478794/563004)   ^MPerforming
 inexact rename detection:  86% (486012/563004)   ^MPerforming inexact rename detection:  87% (490824/563004)   ^MPerforming 
inexact rename detection:  88% (495636/563004)   ^MPerforming inexact rename detection:  89% (502854/563004)   ^MPerforming i
nexact rename detection:  90% (507666/563004)   ^MPerforming inexact rename detection:  91% (512478/563004)   ^MPerforming in
exact rename detection:  92% (519696/563004)   ^MPerforming inexact rename detection:  93% (524508/563004)   ^MPerforming ine
xact rename detection:  94% (529320/563004)   ^MPerforming inexact rename detection:  95% (536538/563004)   ^MPerforming inex
act rename detection:  96% (541350/563004)   ^MPerforming inexact rename detection:  97% (546162/563004)   ^MPerforming inexa
ct rename detection:  98% (553380/563004)   ^MPerforming inexact rename detection:  99% (558192/563004)   ^MPerforming inexac
t rename detection: 100% (563004/563004)   ^MPerforming inexact rename detection: 100% (563004/563004), done.
 
2020-12-14 07:39:49,265 8208 INFO mergebot odoo.addons.forwardport.cherrypick.61199: e5ccb69f7a08fe5a9a66fc2888448dc2b5428f8f
: success -> a886d576f5c56ddb5d77ffb65ae037ad61366ec3
```

**Comment by xmo-odoo** [comment]
Also might be worth filtering out the inexact rename detection thing before printing the stdout (see CherrypickError below).
